aws/resource_aws_xray_sampling_rule_test.go

```diff
@@ -0,0 +1,209 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/xray"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSXraySamplingRule_basic(t *testing.T) {
+	var samplingRule xray.SamplingRule
+	resourceName := "aws_xray_sampling_rule.test"
+	rString := acctest.RandString(8)
+	ruleName := fmt.Sprintf("tf_acc_sampling_rule_%s", rString)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSXraySamplingRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSXraySamplingRuleConfig_basic(ruleName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckXraySamplingRuleExists(resourceName, &samplingRule),
+					testAccCheckResourceAttrRegionalARN(resourceName, "rule_arn", "xray", fmt.Sprintf("sampling-rule/%s", ruleName)),
+					resource.TestCheckResourceAttrSet(resourceName, "priority"),
```

Comment:
For these `TestCheckResourceAttrSet` checks here we should prefer to actually check the value in the state to ensure it is what we expect from the configuration, e.g.

```go
resource.TestCheckResourceAttr(resourceName, "priority", "5"),
// ... repeat for below attributes ...
```